examples/crd_derive_schema.rs

```diff
@@ -0,0 +1,275 @@
+                                              "nullable": true,
+                                              "type": "string"
+                                          },
+                                          "nullable_with_default": {
+                                              "default": "default_nullable",
+                                              "nullable": true,
+                                              "type": "string"
+                                          },
+                                          "nullable_with_default_no_skip": {
+                                              "default": "default_nullable",
+                                              "nullable": true,
+                                              "type": "string"
+                                          },
+                                          "required_field": {
+                                              "type": "string"
+                                          },
+                                          "with_default": {
+                                              "default": "default_value",
+                                              "type": "string"
+                                          },
+                                        },
+                                        "required": [
+                                            "required_field"
+                                        ],
+                                        "type": "object"
+                                    }
+                                },
+                                "required": [
+                                    "spec"
+                                ],
+                                "title": "Foo",
+                                "type": "object"
+                            }
+                        },
+                        "subresources": {},
+                    }
+                ]
+            }
+        });
+        let expected = serde_json::from_value(output).unwrap();
+        // println!("{}", serde_yaml::to_string(&crd).unwrap());
+        // println!("{}", serde_yaml::to_string(&expected).unwrap());
+        assert_eq!(crd, expected);
+    }
+
+    // All tests are in this test case to avoid multiple test cases from creating/deleting Foo.
+    #[tokio::test]
+    async fn test_create() {
+        // Creating CRD v1 works as expected.
+        let client = Client::try_default().await.unwrap();
+        delete_crd(client.clone()).await.unwrap();
+        assert!(create_crd(client.clone()).await.is_ok());
+
+        // Test creating Foo resource.
+        let namespace = std::env::var("NAMESPACE").unwrap_or("default".into());
+        let foos = Api::<Foo>::namespaced(client.clone(), &namespace);
+        let bar = Foo::new(
+            "bar",
+            FooSpec {
+                required_field: "required".into(),
+                with_default: "".into(),
```

Comment:
This test case actually elides the defaulting with the explicit empty string. Is this intended?
Better to use `..Default()::default()`.

Note to self: Oh, I guess this is why you want untyped values - so you can skip sending them, and check that defaulting occurs via schema. Yeah, patch is useful for that.